Makefile

```diff
@@ -34,6 +36,28 @@ install: all
 	ln -rsf ${VKDTDIR}/vkdt $(DESTDIR)$(prefix)/bin/vkdt
 	ln -rsf ${VKDTDIR}/vkdt-cli $(DESTDIR)$(prefix)/bin/vkdt-cli
 
+libinstall: ext lib bin
+	mkdir -p $(VKDTLIBDIR)
+	mkdir -p $(VKDTLIBDIR)/modules
+	mkdir -p $(VKDTLIBDIR)/data
+	mkdir -p $(VKDTINCDIR)
+	mkdir -p $(VKDTINCDIR)/qvk
+	mkdir -p $(VKDTINCDIR)/pipe
+	mkdir -p $(VKDTINCDIR)/pipe/modules
+	mkdir -p $(VKDTINCDIR)/core
+	mkdir -p $(VKDTINCDIR)/gui
+	cp -rfL bin/data ${VKDTLIBDIR}
+	rsync -avP --include='**/params' --include='**/connectors' --include='**/*.ui' --include='**/ptooltips' --include='**/ctooltips' --include='**/readme.md' --include='**.spv' --include='**.so' --include '*/' --exclude='**' bin/modules/ ${VKDTLIBDIR}/modules/
+	cp -rfL bin/libvkdt.so ${VKDTLIBDIR} 
+	cp -rfL bin/libvkdt.h $(VKDTINCDIR)
+	cp -rfL src/qvk/*.h $(VKDTINCDIR)/qvk
+	cp -rfL src/pipe/*.h $(VKDTINCDIR)/pipe
+	cp -rfL src/pipe/modules/*.h $(VKDTINCDIR)/pipe/modules
+	cp -rfL src/gui/*.h $(VKDTINCDIR)/gui
+	cp -rfL src/core/*.h $(VKDTINCDIR)/core
+	echo "$(VKDTLIBDIR)" > /etc/ld.so.conf.d/libvkdt.conf
```

Comment:
oh! root make rules! i'd probably either not do this at all (leave it to the user to either install to where ldd can find it or take care of this in a distro-specific way) or maybe for convenience provide this as a separate rule.

if people build packages, they would set DESTDIR to something say /tmp/package-version/ and do like a test install to grab all the installed files from later on, to put in the package. if the install make target also messes with the global system configuration this becomes messy.

Reply:
I'm learning this as I go, but it looks to me ldconfig has to be involved in the dlopen/dlinfo dance below, and this is how it's set up.

Reply:
well yes for the system to pick it up automatically you need to rerun ldconfig. that's done after installing packages, and people who install libraries to their system usually remember doing it i would assume. also the `/etc/ld.so.conf.d/` way of doing things may or may not be a universal thing on all distros. works on debian (tm).

i'm more worried about the separation of concerns here. again, maybe you just want to try an install to a non-zero DESTDIR to collect all files into a tarball. you don't want that install rule to modify your host system (especially not with a library path pointing to /tmp/something...).